Hannah-Sten/TeXiFy-IDEA
# Fix "Move Section Contents to Seperate File" intention

Fix #2724 

#### Summary of additions and changes

* Fixes bug outlined in #2724 
* Changed the semantics behind `nl.hannahsten.texifyidea.inspections.latex.codestyle.LatexTooLargeSectionInspection#findNextSection(LatexCommands)` to require a section/chapter. I don't think it is the *best*, but I am unsure what else it is used for. If the function is only used to find the next section, from a section, then there will be no issue.
* Note there is no guard to verify that the passed command is actually a valid section. I did this because I wanted to steer away from backsearching for one, since that requires more intimate knowledge of the usage than a basic "Find Usages" will provide.

#### How to test this pull request

```latex
\documentclass[11pt]{article}

\usepackage{amsmath}

\begin{document}

    \chapter{Foo}


    \section{SubFoo}
    Bar

    \subsubsection{SubbFoo}
    Barr

    \section{SubFooToo}


    \chapter{Too}


    \section{SubToo}
    \begin{equation}
        mathisfun
    \end{equation}

\end{document}
```

To test, take the above test file, and use the `Move Section Contents to new File` inspection. Perform this on every section and chapter and verify it works as expected. After running the inspection, undo the inspection, then try a second one. After each one works individually, try combinations thereof.
